packages/cli-repl/test/e2e-fle.spec.ts

```diff
@@ -299,9 +299,117 @@ describe('FLE tests', () => {
+              ]
+            },
+            {
+              _id: 'ghjk',
+              v: Binary(Buffer.from("072871cd1d83174d0c92be1ac934ed26b10299f34210f149673b61f0d369f89290577c410b800ff38ed10eec235aef3677d3594c6371dd5b8f8d4c34769228bf7aea00b1754036a5850a4fef25c40969451151695614ae6142e954bab6c72080b5f43ccac774f6a1791bcc2ca4ca8998b9d5148441180631c7d8136034ff5019ca31a082464ec2fdcf212460a121d14dec3b8ee313541dc46689c79636929f0828dfdef7dfd4d53e1a924bbf70be34b1668d9352f6102a32265ec45d9c5cc0d7cf5f9266cf161497ee5b4a9495e16926b09282c6e4029d22d88e", "hex"), 6),
+              __safeContent__: [
+                Binary(Buffer.from("b04e26633d569cb47b9cbec650d812a597ffdadacb9a61ee7b1661f52228d661", "hex"), 0)
+              ]
+            }
+          ],
+          bypassDocumentValidation: true
+        });
+
+        db.runCommand({
+          insert: 'enxcol_.encryptiontest.ecoc',
+          documents: [
+            {
+              _id: ObjectId("6454e14689ef42f381f7336b"),
+              fieldName: 'v',
+              value: Binary(Buffer.from("3eb89d3a95cf955ca0c8c56e54018657a45daaf465dd967d9b24895a188d7e3055734f3c0af88302ceab460874f3806fe52fa4541c9f4b32b5cee6c5a6df9399da664f576dd9bde23bce92f5deea0cb3", "hex"), 0)
+            },
+            {
+              _id: ObjectId("6454e14689ef42f381f73385"),
+              fieldName: 'v',
+              value: Binary(Buffer.from("2299cd805a28efb6503120e0250798f1b19137d8234690d12eb7e3b7fa74edd28e80c26022c00d53f5983f16e7b5abb7c3b95e30f265a7ba36adb290eda39370b30cedba960a4002089eb5de2fd118fc", "hex"), 0)
+            }
+          ],
+          bypassDocumentValidation: true
+        });
+
+        db.runCommand({
+          insert: 'enxcol_.encryptiontest.esc',
+          documents: [
+            {
+              _id: Binary(Buffer.from("51db700df02cbbfa25498921f858d3a9d5568cabb97f7283e7b3c9d0e3520ac4", "hex"), 0),
+              value: Binary(Buffer.from("dc7169f28df2c990551b098b8dec8f5b1bfeb65d3f40d0fdb241a518310674a6", "hex"), 0)
+            },
+            {
+              _id: Binary(Buffer.from("948b3d29e335485b0503ffc6ade6bfa6fce664c2a1d14790a523c09223da3f09", "hex"), 0),
+              value: Binary(Buffer.from("7622097476c59c0ca5bf9d05a52fe725517e03ad811f6c073b0d0184a9d26131", "hex"), 0)
+            }
+          ],
+          bypassDocumentValidation: true
+        });
+
+        // now set up a new client with QE so we can test the automatic decryption
+        client = Mongo(${JSON.stringify(uri)}, {
+          keyVaultNamespace: '${dbname}.keyVault',
+          kmsProviders: { local: { key: 'A'.repeat(128) } }
+        });
+        coll = client.getDB('${dbname}').encryptiontest;
+      }`
+      );
+      expect(await shell.executeLine('({ count: coll.countDocuments() })')).to.include('{ count: 2 }');
+
+      // Make sure the find payload allows searching for the encrypted value
+      expect(await shell.executeLine('coll.findOne({ v: "456" })._id')).to.include('ghjk');
```

Comment:
This does not find the row, but mongosh does decrypt the data when doing a find() with no filter.